optuna_dashboard/static/components/GraphContour.tsx

```diff
@@ -0,0 +1,280 @@
+  )
+
+  let paramNames = new Set<string>(trials[0].params.map((p) => p.name))
+  filteredTrials.forEach((t) => {
+    paramNames = new Set<string>(
+      t.params.filter((p) => paramNames.has(p.name)).map((p) => p.name)
+    )
+  })
+
+  const paramnames = Array.from(paramNames)
+
+  const classes = useStyles()
+  const [objectiveId, setObjectiveId] = useState<number>(0)
+  const [xAxis, setXAxis] = useState<string>(paramnames[0])
+  const [yAxis, setYAxis] = useState<string>(paramnames[1])
+
+  const handleObjectiveChange = (
+    event: React.ChangeEvent<{ value: unknown }>
+  ) => {
+    setObjectiveId(event.target.value as number)
+  }
+
+  const handleXAxisChange = (e: ChangeEvent<{ value: unknown }>) => {
+    setXAxis(e.target.value as string)
+  }
+
+  const handleYAxisChange = (e: ChangeEvent<{ value: unknown }>) => {
+    setYAxis(e.target.value as string)
+  }
+
+  useEffect(() => {
+    if (study !== null) {
+      plotContour(study, objectiveId, xAxis, yAxis)
+    }
+  }, [study, objectiveId, xAxis, yAxis])
+
+  return (
+    <Grid container direction="row">
+      <Grid item xs={3}>
+        <Grid container direction="column">
+          {study !== null && study.directions.length !== 1 ? (
+            <FormControl component="fieldset" className={classes.formControl}>
+              <FormLabel component="legend">Objective ID:</FormLabel>
+              <Select value={objectiveId} onChange={handleObjectiveChange}>
+                {study.directions.map((d, i) => (
+                  <MenuItem value={i} key={i}>
+                    {i}
+                  </MenuItem>
+                ))}
+              </Select>
+            </FormControl>
+          ) : null}
+          <FormControl component="fieldset" className={classes.formControl}>
+            <InputLabel id="parameter1">X Axis Parameter</InputLabel>
+            <Select value={xAxis} onChange={handleXAxisChange}>
+              {paramnames.map((x) => (
+                <MenuItem value={x} key={x}>
+                  {x}
+                </MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+          <FormControl component="fieldset" className={classes.formControl}>
+            <InputLabel id="parameter2">Y Axis Parameter</InputLabel>
+            <Select value={yAxis} onChange={handleYAxisChange}>
+              {paramnames.map((x) => (
+                <MenuItem value={x} key={x}>
+                  {x}
+                </MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+        </Grid>
+      </Grid>
+      <Grid item xs={6}>
+        <div id={plotDomId} />
+      </Grid>
+    </Grid>
+  )
+}
+
+const plotContour = (
+  study: StudyDetail,
+  objectiveId: number,
+  xAxis: string,
+  yAxis: string
+) => {
+  if (document.getElementById(plotDomId) === null) {
+    return
+  }
+
+  const layout: Partial<plotly.Layout> = {
+    title: "Contour",
+    margin: {
+      l: 50,
+      r: 50,
+    },
+  }
```

Comment:
Could you explain why this is different with `updateLayout`?

Reply:
I too don't find much use of using two different variables `layout` and `updateLayout`. I'll avoid unnecessary code repitition from next time. Also, I'll make the necessary change.